perf_test/CMakeLists.txt

```diff
@@ -50,5 +50,97 @@ if (KokkosKernels_ENABLE_PERFTESTS)
     ADD_COMPONENT_SUBDIRECTORY(blas)
     ADD_SUBDIRECTORY(performance)
     #ADD_SUBDIRECTORY(common)
+
+    IF (KOKKOSKERNELS_HAS_TRILINOS)
+      message(FATAL_ERROR "Benchmarks are not supported when building as part of Trilinos")
+    ENDIF()
+
+    find_package(benchmark QUIET)
+    IF(benchmark_FOUND)
+      MESSAGE(STATUS "Using google benchmark found in ${benchmark_DIR}")
+    ELSE()
+      message(STATUS "No installed google benchmark found, fetching from GitHub")
+      include(FetchContent)
+      SET(BENCHMARK_ENABLE_TESTING OFF)
+
+      list(APPEND CMAKE_MESSAGE_INDENT "    ")
+      #Note: recent bug (google/benchmark#1441) is preventing us from using
+      # the latest benchmark release.
+      FetchContent_Declare(
+        googlebenchmark
+        URL https://github.com/google/benchmark/archive/refs/tags/v1.6.2.tar.gz
+        URL_HASH MD5=14d14849e075af116143a161bc3b927b
+      )
+      FetchContent_MakeAvailable(googlebenchmark)
+      list(POP_BACK CMAKE_MESSAGE_INDENT)
+
+      include_directories(${benchmark_SOURCE_DIR}/include)
+
+      # Suppress clang-tidy diagnostics on code that we do not have control over
+      IF(CMAKE_CXX_CLANG_TIDY)
+        SET_TARGET_PROPERTIES(benchmark PROPERTIES CXX_CLANG_TIDY "")
+      ENDIF()
+
+      target_compile_options(benchmark PRIVATE -w)
+      target_compile_options(benchmark_main PRIVATE -w)
+    ENDIF()
+
+
+    FUNCTION(KOKKOSKERNELS_ADD_BENCHMARK NAME)
+      CMAKE_PARSE_ARGUMENTS(
+        BENCHMARK
+        ""
+        ""
+        "SOURCES"
+        ${ARGN}
+      )
+      IF(DEFINED BENCHMARK_UNPARSED_ARGUMENTS)
+        MESSAGE(
+          WARNING
+          "Unexpected arguments when adding a benchmark: "
+          ${BENCHMARK_UNPARSED_ARGUMENTS}
+        )
+      ENDIF()
+
+      SET(BENCHMARK_NAME ${PACKAGE_NAME}_${NAME})
+
+      ADD_EXECUTABLE(
+        ${BENCHMARK_NAME}
+        ${BENCHMARK_SOURCES}
+      )
+      TARGET_LINK_LIBRARIES(
+        ${BENCHMARK_NAME}
+        PRIVATE benchmark::benchmark Kokkos::kokkoskernels
+      )
+      FOREACH(SOURCE_FILE ${BENCHMARK_SOURCES})
+        SET_SOURCE_FILES_PROPERTIES(
+          ${SOURCE_FILE}
+          PROPERTIES LANGUAGE CXX
+        )
+      ENDFOREACH()
+
+      STRING(TIMESTAMP BENCHMARK_TIME "%Y-%m-%d_T%H-%M-%S" UTC)
+      SET(
+        BENCHMARK_ARGS
+        --benchmark_counters_tabular=true
+        --benchmark_out=${BENCHMARK_NAME}_${BENCHMARK_TIME}.json
+      )
+
+      ADD_TEST(
```

Comment:
Will this be run with all other unit tests?

Reply:
No this would probably only be triggered in nightly build or even at a different frequency. The benchmark might be longer and could time out if included next to unit-tests

Reply:
That will be run with all other unit tests only if KokkosKernels_ENABLE_BENCHMARK is enabled, which is not the case by default